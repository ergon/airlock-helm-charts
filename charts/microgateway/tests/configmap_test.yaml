suite: test configmap
release:
  name: my-microgateway
templates:
- templates/configmap.yaml

tests:
  - it: should be ConfigMap
    asserts:
      - isKind:
          of: ConfigMap

  - it: dsl example
    values:
      - ./values/dsl.yaml
    asserts:
      - matchSnapshot:
          path: data.[config.yaml]

  - it: should be no configmap if one is provided
    set:
      config:
        dslConfigMap: testmap
    asserts:
      - hasDocuments:
          count: 0  
