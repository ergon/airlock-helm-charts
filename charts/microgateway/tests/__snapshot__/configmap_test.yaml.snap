dsl example:
  1: |
    |-
      apps:
      - mappings:
        - backend:
            hosts:
            - name: dsl-backend
              port: 8080
              protocol: http
          entry_path: /dsl/
          name: echo
        virtual_host:
          hostname: echo
